Sources/PackageModel/Destination.swift

```diff
@@ -0,0 +1,120 @@
+import Foundation
+import Basic
+
+enum HostDestinationError: Swift.Error {
+    /// Couldn't find the Xcode installation.
+    case invalidInstallation(problem: String)
+}
+
+extension HostDestinationError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .invalidInstallation(let problem):
+            return problem
+        }
+    }
+}
+
+enum DestinationFileError: Swift.Error {
+    /// Wrong version.
+    case destinationFileVersionMismatch(problem: String)
+}
+
+extension DestinationFileError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .destinationFileVersionMismatch(let problem):
+            return problem
+        }
+    }
+}
+
+public struct Destination {
+    public var target: String
+
+    public var sdk: AbsolutePath
+
+    public var toolchain: AbsolutePath
+
+    public var dynamicLibraryExtension: String
+
+    public var extraCCFlags: [String]
+
+    public var extraSwiftCFlags: [String]
```

Comment:
yes, but it's a value type so changes only affect the very local variable. I'm happy to make it `let` but don't think it makes much difference because it's a `struct`. Let me know what you think.